tests/zfs-tests/tests/functional/mmp/mmp_on_uberblocks.ksh

```diff
@@ -37,43 +40,43 @@
 verify_runnable "both"
 
 UBER_CHANGES=0
+EXPECTED=$(($(echo $DISK | wc -w) * 10))
+FUDGE=$((EXPECTED * 20 / 100))
+MIN=$((EXPECTED - FUDGE))
+MAX=$((EXPECTED + FUDGE))
+KSTAT_PATH="/proc/spl/kstat/zfs/$TESTPOOL/multihost"
 
 function cleanup
 {
 	default_cleanup_noexit
 	set_tunable64 zfs_txg_timeout $TXG_TIMEOUT_DEFAULT
-	log_must rm -f $PREV_UBER $CURR_UBER
+	log_must set_tunable64 zfs_multihost_history $MMP_HISTORY
 	log_must mmp_clear_hostid
 }
 
 log_assert "Ensure MMP uberblocks update at the correct interval"
 log_onexit cleanup
 
 log_must set_tunable64 zfs_txg_timeout $TXG_TIMEOUT_LONG
+log_must set_tunable64 zfs_multihost_history $MMP_HISTORY_OFF
 log_must mmp_set_hostid $HOSTID1
 
 default_setup_noexit $DISK
+log_must set_tunable64 zfs_multihost_history $MMP_HISTORY
 log_must zpool set multihost=on $TESTPOOL
 
-log_must zdb -u $TESTPOOL > $PREV_UBER
+log_must sleep 10
 
-SECONDS=0
-while [[ $SECONDS -le 10 ]]; do
-	log_must zdb -u $TESTPOOL > $CURR_UBER
-	if ! diff -u "$CURR_UBER" "$PREV_UBER"; then
-		(( UBER_CHANGES = UBER_CHANGES + 1 ))
-		log_must mv "$CURR_UBER" "$PREV_UBER"
-	fi
-done
+UBER_CHANGES=$(cat "$KSTAT_PATH" | sed '1,2d' | wc -l)
```

Comment:
A helper function in `mmp.kshlib` may make sense here since again mmp_reset_interval.ksh and mmp_write_uberblocks.ksh are doing something verify similar.